astropy/wcs/wcs.py

```diff
@@ -909,10 +909,12 @@ def _read_det2im_kw(self, header, fobj, err=0.0):
         try:
             axiscorr = header["AXISCORR"]
             d2imdis = self._read_d2im_old_format(header, fobj, axiscorr)
```

Comment:
Only one of these should be in the `try`-statement, probably the `axiscorr=`, and the other should be moved to the `else` block.